### ISSUE TITLE
Feature request: optional skip OCSP checks (certbot list cetificates)

### ISSUE THREAD

**Issue by guard43ru** [certbot/certbot#9553]
## My operating system is (include version):
Debian 11.6

## I installed Certbot with (snap, OS package manager, pip, certbot-auto, etc):
apt (certbot 1.12.0)

## Feature request
Hello, we have large setup (use 1k+ LE certs), and we have problem with performance.
When I just want to get list of certificates:
```
$ time certbot certificates
Saving debug log to /var/log/letsencrypt/letsencrypt.log
...

real	4m25,679s
user	0m34,075s
sys	0m1,906s
```
OCSP checks take most of the time, and these checks are not always necessary.
In old PR discussed this option "--check-ocsp newer" https://github.com/certbot/certbot/pull/3941#issuecomment-268381594 but in the end it wasn't accepted.

**Comment by alexzorin** [comment]
Firing off 1k+ HTTP requests every time somebody runs `certbot certificates` isn't great. Besides adding `--no-check-revocation`, maybe we can add something clever to the logic of `certbot certificates`. Like, disabling revocation checking if the number of certificates is too large.

Leaving this in triage to get input from others.

**Comment by bmw** [comment]
Yeah I agree this isn't great and I'm open to something being done here.

I'd personally be fine with either approach honestly as long as there is a way for people with large number of certificates to get the OCSP status of at least a subset of their certificates from Certbot.

**Comment by osirisinferi** [comment]
> as long as there is a way for people with large number of certificates to get the OCSP status of at least a subset of their certificates from Certbot.

Any thoughts about how Certbot would decide which subset to check and which subset to ignore?

**Comment by bmw** [comment]
> Any thoughts about how Certbot would decide which subset to check and which subset to ignore?

`certbot certificates` already accepts both `--cert-name` and `--domains` and limits its output to matching certificates.

I think the design I personally like best is to somewhat arbitrarily pick a number of certificates above which `certbot certificates` won't do OCSP checks by default. If we hit that threshold, I think in the output of that command we should include something about how the OCSP checks were skipped due to the number of certificates and if the user wants to see the OCSP status, they should use `--cert-name` or `--domains` to select a subset of their certificates. (We could also add a flag like `--force-revocation-check` or `--force-ocsp-check`, but I'm not sure it's really needed.)

> Also, should OCSP also be skipped when running certbot renew?

My opinion here is no, we shouldn't even give the option for people to disable OCSP checks with `certbot renew`. That command is usually run non-interactively so I think the extra time shouldn't matter much and I think its a dangerous footgun that could cause people to have issues they otherwise wouldn't have.

**Comment by alexzorin** [comment]
Purely out of nerd curiosity, I had a look at what the performance different might be [if we `async`'d the crap out of the OCSP checks](https://github.com/certbot/certbot/compare/master...alexzorin:async-ocsp) in `certbot certificates`.

With 1000 staging certificates on `async` with default settings on my laptop:

    sudo certbot certificates  4.63s user 0.43s system 90% cpu 5.599 total

without:

    sudo certbot certificates  16.38s user 1.98s system 36% cpu 50.022 total

I doubt it's worth doing but interesting nonetheless.

>My opinion here is no, we shouldn't even give the option for people to disable OCSP checks with `certbot renew`.

On this, maybe we should follow [Erica's suggestion](https://github.com/certbot/certbot/pull/3941#discussion_r93356455) to call it `--no-network` or `--offline` or something similar rather than `--no-check-revocation`, so that it does not give the suggestion that it can be used with `certbot renew`.

**Comment by guard43ru** [comment]
> we shouldn't even give the option for people to disable OCSP checks with `certbot renew`. That command is usually run non-interactively

I agree, we don't have problem with `renew` (it's automated).

> I think the design I personally like best is to somewhat arbitrarily pick a number of certificates above which `certbot certificates` won't do OCSP checks by default.

Yes it's can help in these case.

**Comment by cocorossello** [comment]
This is a problem for us, just reading the certificates takes more than minutes. 

A little workaround is mapping the host  **r3.o.lencr.org** to some internal network host like 192.168.130.44 so the OCSP always get a 404 error (anyway I don't care at that point).  With this workaround we can decrease the time from around 2 minutes 20 seconds to around 40 seconds

**Comment by SergeyMosin** [comment]
#### Let’s Encrypt Ending OCSP Support in 2025 ( https://letsencrypt.org/2024/12/05/ending-ocsp/ )

> August 6, 2025
> - On this date we will turn off our OCSP responders

Are there any plans for this ?
